### PR TITLE
refactor(ng-ovh-payment-method): rework payment method implementation

### PR DESCRIPTION
**Pull request by Jisay**
# :boom: Breaking changes
- removal of `ovhPaymentMethodProvider`. This provider only provided a way to set `target`. As the module now use `@ovh-ux/manager-core` as dependencies `target` is now available through 
`@ovh-ux/manager-core` region ;
- removal of `ovhPaymentMethodAvailableTypes`. You can now use `ovhPaymentMethodRegister` component.

# :rocket: Feature
- create a set of directives usable on any page of the manager for payment method registration.

# :recycle: Refactor
- use class with methods to use payment methods params easily ;
- add enums and define them as angular constants ;
- depreciate some methods...

# :house_with_garden: Internal
- PAYM-248

### CODE REVIEW

File: packages/components/ng-ovh-payment-method/src/components/integration/constants.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  PENDING: 'pending',
+  SUCCESS: 'success',
+};
+
+export const DEFAULT_BINDINGS_VALUES = {
+  callbackStatusParamUrlName: 'status',
+};
+
+export default {
+  DEFAULT_BINDINGS_VALUES,
```
Comment (antleblanc) [inline packages/components/ng-ovh-payment-method/src/components/integration/constants.js]: Both `const` can also be exported in `default` I guess:

```suggestion
  AVAILABLE_CALLBACK_STATUS_ENUM,
  DEFAULT_BINDINGS_VALUES,
```

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#6035.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#6035.1 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#6036.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#6127.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#6127.2 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#6142.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build-monorepository#6142.1 ✘  
* 
  * Build ✘

**Pull request merged**